lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_handlers.cpp

```diff
@@ -1833,11 +1839,9 @@ sgw_eps_bearer_context_information_t* update_sgw_context_to_s11_teid_map(
   /* Once sgw_s8_teid is obtained from orc8r, move sgw_eps_bearer_context
    * from temporary_create_session_procedure_id hashlist  to sgw_teid hashlist
    */
-  sgw_eps_bearer_context_information_t* sgw_context_p = NULL;
-  hashtable_ts_remove(
-      sgw_state->temporary_create_session_procedure_id_htbl,
-      (const hash_key_t)session_rsp_p->temporary_create_session_procedure_id,
-      (void**)&sgw_context_p);
+  sgw_eps_bearer_context_information_t* sgw_context_p = nullptr;
+  sgw_state->temporary_create_session_procedure_id_map.get(
```

Comment:
What's the purpose of getting this entry before removing it?

Reply:
While handling create session request, sgw_context is created and inserted into map,  temporary_create_session_procedure_id_map. Because for roaming users, sgw_s11_teid would be allocated by external PGW, shall be received in create session response message. So on reception of create session response, sgw_context is removed from map, but not freed. 
So first we get the sgw_context, remove from map, temporary_create_session_procedure_id_map and insert into state_teid_map